solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+    {
+        _networkToken = networkToken;
+        _govTokenGovernance = govTokenGovernance;
+        _govToken = govTokenGovernance.token();
+
+        _burnIncentiveFee = burnIncentiveFee;
+        _maxBurnIncentiveFeeAmount = maxBurnIncentiveFeeAmount;
+    }
+
+    /**
+     * @dev ETH receive callback
+     */
+    receive() external payable {}
+
+    /**
+     * @dev returns the burn incentive fee and its maximum amount
+     *
+     * @return the burn incentive fee and its maximum amount
+     */
+    function burnIncentiveFee() external view returns (uint32, uint256) {
+        return (_burnIncentiveFee, _maxBurnIncentiveFeeAmount);
+    }
+
+    /**
+     * @dev allows the owner to set the burn incentive fee and its maximum amount
+     *
+     * @param newBurnIncentiveNetworkFee the percentage of the converted network tokens to be sent to the caller of the vortex (in units of PPM)
+     * @param newMaxBurnIncentiveFeeAmount the maximum incentive fee to be sent to the caller of the vortex
+     */
+    function setBurnIncentiveFee(uint32 newBurnIncentiveNetworkFee, uint256 newMaxBurnIncentiveFeeAmount)
+        external
+        ownerOnly
+        validFee(newBurnIncentiveNetworkFee)
+    {
+        emit BurnIncentiveFeeUpdated(
+            _burnIncentiveFee,
+            newBurnIncentiveNetworkFee,
+            _maxBurnIncentiveFeeAmount,
+            newMaxBurnIncentiveFeeAmount
+        );
+
+        _burnIncentiveFee = newBurnIncentiveNetworkFee;
+        _maxBurnIncentiveFeeAmount = newMaxBurnIncentiveFeeAmount;
+    }
+
+    /**
+     * @dev returns the total amount of the burned governance tokens
+     *
+     * @return total amount of the burned governance tokens
+     */
+    function totalBurnedAmount() external view returns (uint256) {
+        return _totalBurnedAmount;
+    }
+
+    /**
+     * @dev converts the provided tokens to governance tokens and burns them
+     *
+     * @param tokens the tokens to convert
+     */
+    function vortex(IERC20[] calldata tokens) external protected {
+        ITokenHolder feeWallet = networkFeeWallet();
+
+        // retrieve conversion paths and the amounts to burn
+        (address[][] memory paths, uint256[] memory amounts, address[] memory govPath) =
+            vortexStrategy(tokens, feeWallet);
+
+        // withdraw all token/ETH amounts to the contract
+        feeWallet.withdrawTokensMultiple(tokens, address(this), amounts);
+
+        // convert all amounts to the network token and record conversion amounts
+        BancorNetwork network = bancorNetwork();
+
+        uint256[] memory networkTokenConversionAmounts = new uint256[](tokens.length);
+        uint256 grossNetworkTokenConversionAmount = 0;
+        uint256 totalGovTokenAmountToBurn = 0;
+
+        for (uint256 i = 0; i < paths.length; ++i) {
+            // avoid empty conversions
+            uint256 amount = amounts[i];
+            if (amount == 0) {
+                continue;
```

Comment:
Why not filter these out in `vortexStrategy`?

Reply:
Yes, good idea